### ISSUE TITLE
[FIRParser] Should produce an error when an expression is not inside a module

### ISSUE THREAD

**Issue by youngar** [llvm/circt#2787]
When the last line is not indented far enough, it is silently discarded.  The parser should produce an error:
```scala
circuit Foo:
  module Foo :
    output io : UInt<1>
  io is invalid
```
`firtool --parse-only test.fir` gives:
```mlir
firrtl.module @Foo(out %io: !firrtl.uint<1>) {
}
```

**Comment by seldridge** [comment]
The rule here, which is documented in the FIRRTL spec, is that the indentation after an indented line establishes the "indent level". After that, all lines (in the same block) need to be at the same level.

This should error out because `module` or `extmodule` are the only legal keywords at that indent level.

The SFC error isn't great here (as the parser is generated), but it gets the point across.  Only `module`, `extmodule`, or a further reduction in indentation are allowed:

```
# firrtl -i Foo.fir
line 4:2 extraneous input 'io' expecting {'module', 'extmodule', DEDENT}
```
